### PR TITLE
concentrators

### PR DESCRIPTION
**Pull request by vanillaHill**
Claim/fix bug issues https://github.com/FortressFinance/issues/issues/232

### CODE REVIEW

File: src/components/Asset/AssetLogo.tsx [hunk 1]
```diff
@@ -27,10 +27,8 @@ export const AssetLogo: FC<AssetLogoProps> = ({ className, tokenAddress }) => {
   const [isError, setIsError] = useState(false)
 
   useEffect(() => {
-    if (!isError) {
-      setIsError(tokenAddress === "0x" || tokenAddress === undefined)
-    }
-  }, [isError, tokenAddress])
+    setIsError(tokenAddress === "0x" || tokenAddress === undefined)
```
Comment (aquaticone) [inline src/components/Asset/AssetLogo.tsx:30]: This doesn't need to be handled with state. Let's leave `isError` for handling actual errors with the image loading:

```tsx
const isBadProps = tokenAddress === "0x" || tokenAddress === undefined
// ...
{isError || isBadProps ? <>error code</> : <>success code</>} 
```

File: src/components/Concentrator/ConcentratorRewards.tsx [hunk 2]
```diff
@@ -207,13 +210,15 @@ const ConcentratorRewardsBalance: FC<ConcentratorRewardsProps> = ({
...
   const rewardsBalance = useConcentratorPendingReward({
-    concentratorAddress: concentrator.data?.ybTokenAddress,
+    ybTokenList: [concentrator.data?.ybTokenAddress],
   })
   const rewardToken = useTokenOrNative({
     address: concentrator.data?.rewardTokenAddress,
   })
   const formatted = ethers.utils.formatUnits(
-    rewardsBalance.data ?? BigNumber.from(0),
+    rewardsBalance.data !== undefined && rewardsBalance.data.length > 0
```
Comment (aquaticone) [inline src/components/Concentrator/ConcentratorRewards.tsx:219]: `rewardsBalance.data?.[0] ?? BigNumber.from(0)`

File: src/hooks/useConcentratorRewards.ts [hunk 3]
```diff
@@ -1,47 +1,64 @@
...
   const { address: userAddress } = useAccount()
-  return useContractRead({
-    address,
-    chainId,
-    abi: AMMConcentratorBase,
-    functionName: "pendingReward",
-    args: [userAddress ?? "0x"],
-    enabled: !!userAddress,
+  const contracts = ybTokenList.map((ybToken) => {
+    return {
```
Comment (aquaticone) [inline src/hooks/useConcentratorRewards.ts:28]:
```ts
ybTokenList.map((ybToken) => ({
  address: ybToken,
  // ...
}))
```

File: src/hooks/useConcentratorRewards.ts [hunk 4]
```diff
@@ -1,47 +1,64 @@
...
+      abi: AMMConcentratorBase,
+      functionName: "pendingReward",
+      args: [userAddress ?? "0x"],
+    }
+  })
+  return useContractReads({
+    contracts: contracts,
+    enabled: !!userAddress && ybTokenList.length > 0,
+    select: (data) =>
+      data.map((reward) => {
```
Comment (aquaticone) [inline src/hooks/useConcentratorRewards.ts:40]:
```ts
data.map((reward) => BigNumber.from(reward ?? 0))
```

File: src/hooks/useConcentratorVault.ts [hunk 5]
```diff
@@ -42,3 +42,33 @@ export function useConcentratorVault({
...
+  primaryAssetList,
+  type,
+}: {
+  targetAsset: Address
+  primaryAssetList: Address[]
+  type?: VaultType
+}) {
+  const registryContract = useRegistryContract()
+  const contracts = primaryAssetList.map((primaryAsset) => {
+    return {
```
Comment (aquaticone) [inline src/hooks/useConcentratorVault.ts:57]: Same here:

```ts
primaryAssetList.map((primaryAsset) => ({
  ...registryContract,
  // ...
}))

File: src/hooks/useConcentratorVault.ts [hunk 6]
```diff
@@ -42,3 +42,33 @@ export function useConcentratorVault({
...
+      ...registryContract,
+      functionName: "getConcentrator",
+      args: [type === "curve", targetAsset ?? "0x", primaryAsset ?? "0x"],
+    }
+  })
+
+  return useContractReads({
+    contracts,
+    select: (data) =>
+      data.map((ybTokenAddress) => {
```
Comment (aquaticone) [inline src/hooks/useConcentratorVault.ts:67]:
```ts
data.map((ybTokenAddress) => (ybTokenAddress ?? "0x") as Address)
```

**Review by aquaticone — APPROVED** [review]
Minor suggestions

**Comment by vanillaHill** [comment]
I will create a new branch after each merge. This time (after your review), I pushed something before seeing your comment, and regrettably, there was a conflict on it, necessitating the merging of the main with my branch and a further push to resolve the issue.

**Pull request merged**